compiler/src/dotty/tools/dotc/core/Names.scala

```diff
@@ -98,7 +98,7 @@ object Names {
     /** A name in the same (term or type) namespace as this name and
      *  with same characters as given `name`.
      */
-    def likeSpaced(name: Name): ThisName
+    def likeSpaced(name: PreName): ThisName
```

Comment:
Currently, there don't seems to be usages for `likeSpaced(PreName)` so I would keep the API as is. If you want to share the implementation with the one in `NameOps`, it seems you can actually replace `PreName` by `Name`